aiohttp_devtools/runserver/main.py

```diff
@@ -49,6 +49,9 @@ def runserver(**config_kwargs: Any) -> RunServer:
 
     main_manager = AppTask(config)
     aux_app.cleanup_ctx.append(main_manager.cleanup_ctx)
```

Comment:
This should already be happening here. Can you try and investigate a bit deeper on the problem?

On shutdown, the cleanup_ctx calls `.close()`, which calls `._stop_dev_server()`.

Reply:
The thing is that I don't see where `cleanup_ctx` is supposed to be called here. [This code](https://github.com/aio-libs/aiohttp-devtools/blob/a9bc32fa6c/tests/test_runserver_main.py#L61) is manually calling `aux_app.on_startup` and `aux_app.on_shutdown` handlers; I did try adding calls to `aux_app._cleanup_ctx._on_startup` and `_on_shutdown`, but that didn't work, I didn't get to investigate further.

Reply:
Aha, so the test code probably needs updating. The main code calls `aiohttp.web.run_app()`, which will run the entire application, including all these handlers.

Reply:
OK, looks like cleanup_ctx is entered on startup, but you need to call `_cleanup()` to exit it.

Reply:
If I haven't confused myself too much, then I think the fix is just to add `await aux_app._cleanup()` after this loop:
https://github.com/aio-libs/aiohttp-devtools/blob/a9bc32fa6c/tests/test_runserver_main.py#L85

Reply:
Unfortunately, `AttributeError: 'Application' object has no attribute '_cleanup'`

Reply:
Ah, sorry, just `cleanup()`:
https://github.com/aio-libs/aiohttp/blob/master/aiohttp/web_app.py#L356

Reply:
That works, thanks! Feel free to squash merge these changes, since the first two commits are now outdated (or I can force push)

Reply:
All our repos are squash merged.